static-assets/components/cstudio-forms/controls/node-selector.js

```diff
@@ -511,8 +511,12 @@ YAHOO.extend(CStudioForms.Controls.NodeSelector, CStudioForms.CStudioFormField,
             }
 
             item.datasource = datasource;
-            this.items[this.items.length] = item;
-
+            if (order) {
```

Comment:
Order = 0 will go to else too. Test specifically for what you need `!= null` (single equals) for example will catch what you seem to want here.